src/Init/Data/AC.lean

```diff
@@ -9,7 +9,7 @@ import Init.Classical
 import Init.ByCases
 
 namespace Lean.Data.AC
-inductive Expr
+inductive Expr where
   | var (x : Nat)
```

Comment:
In #5236 I thought the argument was leaning towards banning `where` for consistency with pattern-matched `def`s.

On the other hand, I guess `match _ with |` is already inconsistent when compared to `inductive _ where |` and `def _ : _ |`...

Reply:
@eric-wieser

> In #5236 I thought the argument was leaning towards banning `where` for consistency with pattern-matched `def`s.

I believe this PR has been superseded by #5542 for approximately this reason.